app/BUILD.bazel

```diff
@@ -281,47 +280,88 @@ ALTERED_DATABINDING_ADAPTERS = [
     "src/main/java/org/oppia/app/databinding/TextViewBindingAdapters.java",
 ]
 
+[DATABINDING_ADAPTERS.append("update_" + altered_adapters[0:-5])
+for altered_adapters in ALTERED_DATABINDING_ADAPTERS]
+
+# TODO(#1617): Remove genrules post-gradle
 '''
-Genrule that alters both resource import statements and class names for adapter files in order to
-account for altered package and file name.
+Genrule for source files in the databinding_adapters library.
+
+Because each databinding library must have a unique package name and manifest, resources must be
+imported using the proper package name when building with Bazel. This genrule alters those imports
+in order to keep Gradle building.
 '''
 [genrule(
      name = "update_" + file[0:-5],
      srcs = [file],
      outs = [file[0:-5] + "_updated.java"],
      cmd = '''
      cat $(SRCS) |
-     sed 's/import org.oppia.app.R/import org.oppia.app.impl.adapters.R/g' |
+     sed 's/import org.oppia.app.R/import org.oppia.app.databinding.adapters.R/g' |
      sed 's/BindingAdapters {/BindingAdapters_updated {/g' > $(OUTS)
      ''',
  )
 for file in ALTERED_DATABINDING_ADAPTERS]
 
 # Source files for recyclerview_adapters library.
-RECYCLERVIEW_ADAPTER_FILES = [
+RECYCLERVIEW_ADAPTERS = [
     "src/main/java/org/oppia/app/recyclerview/RecyclerViewBindingAdapter.java"
 ]
 
-BINDING_ADAPTER_FILES = DATABINDING_ADAPTER_FILES + RECYCLERVIEW_ADAPTER_FILES
+ADAPTERS = DATABINDING_ADAPTERS + RECYCLERVIEW_ADAPTERS
 
 # Library for non-layout resource files
 android_library(
     name = "resources",
     custom_package = "org.oppia.app",
     manifest = "src/main/AndroidManifest.xml",
-    resource_files = glob(["src/main/res/**",], exclude = ["src/main/res/layout*/**"]),
+    resource_files = glob(["src/main/res/**",], exclude = LAYOUTS),
     exports_manifest = True,
     deps = [
         artifact("com.google.android.material:material"),
     ],
     visibility = ["//visibility:private"],
 )
 
+# Library for layout resource files
+android_library(
+    name = "databinding_resources",
+    custom_package = "org.oppia.app.databinding",
+    manifest = "src/main/DatabindingResourcesManifest.xml",
+    srcs = ADAPTERS,
+    resource_files = glob(LAYOUTS),
+    enable_data_binding = 1,
```

Comment:
prefer ``enable_data_binding = True`` here & everywhere for consistency & clarity.

Reply:
Done